src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -207,12 +204,9 @@ public ResultSet.ResultMetadata getResultMetadata()
     TxnNamedRead createNamedRead(NamedSelect namedSelect, QueryOptions options, ClientState state)
     {
         SelectStatement select = namedSelect.select;
-        ReadQuery readQuery = select.getQuery(options, 0);
-        checkTrue(readQuery instanceof  SinglePartitionReadQuery.Group, ILLEGAL_RANGE_QUERY_MESSAGE, select.asCQL(options, state));
-
         // We reject reads from both LET and SELECT that do not specify a single row.
         @SuppressWarnings("unchecked")
-        SinglePartitionReadQuery.Group<SinglePartitionReadCommand> selectQuery = (SinglePartitionReadQuery.Group<SinglePartitionReadCommand>) readQuery;
+        SinglePartitionReadQuery.Group<SinglePartitionReadCommand> selectQuery = (SinglePartitionReadQuery.Group<SinglePartitionReadCommand>) select.getQuery(options, 0);
```

Comment:
why prefer ClassCast rather than `ILLEGAL_RANGE_QUERY_MESSAGE`?

Reply:
In and the other case, the statement at this point is guaranteed to not be partition range query. It has been verified during preparation.